scripts/frame/main.js

```diff
@@ -1,11 +1,40 @@
+/**
+ * @description Used as an alternative `main` to view photos with "frame mode"
+ *
+ * Note, the build script picks a subset of the JS files to build a variant
+ * of the JS code for the special "frame mode".
+ * As this variant does not include all JS files, some objects are missing.
+ * Hence, we must partially re-implement these objects to the extent which is
+ * required by the methods we call.
+ *
+ * This approach is very tedious and error-prone, because we actually
+ * duplicate code.
+ * Also, it is not documented nor obvious why these "subset implementations"
+ * are necessary.
+ * Ideally, the full code base would be used all the time independent of
+ * the users entry point.
+ *
+ * TODO: Find out why we actually need this approach. Re-implementing different variants of the same objects is very error-prone.
```

Comment:
The view mode provides the <meta> tags directly in the html.
In the case of the normal view, because we fetch the data via Ajax, the meta tag can't be set when loading the page (without executing the JS).

Reply:
Thanks @ildyria for the explanation. This gives an argument why there is a difference on the technical level and I will add this explanation to the comment.

Nonetheless I think this sound not like a good justification. To me it sounds more like "someone wanted to hack together a view mode really quickly without thinking about a proper architecture or spending time to rewrite the necessary parts of Lychee and so we ended up with a lot of code duplication". :grinning:

> Yeah, I've wanted get rid of the separate "view" mode since forever.

Me too, now even more after I heard the explanation.

> But, of course, with a new front-end on the horizon, would it make sense to make major changes to the existing one?

Right, same consideration here. I won't spend any time on it.

I would do so if we were sure to keep that frontend. In my mind I have big master plan how the spaghetti code could be re-factored step-by-step (in particular the `view.js` is a mess) and then even replace single pieces by vue.js components gradually. But that's not worth the time, if it is going to die anyway.

Reply:
Comment fixed in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/c376fb51cb7e9e2d8535fd03628ca151ecb4393b

Reply:
This `view` page (and js) is a direct legacy from Tobias code on the v3.

When I built the v4 it was first a direct translation of the v3 into the Laravel framework, then it evolved to make it have more structure.